src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -132,8 +132,18 @@ CATCH_RETURN()
                                                                FLOAT originY)
 {
     const auto drawingContext = static_cast<const DrawingContext*>(clientDrawingContext);
-    _formatInUse = drawingContext->useItalicFont ? _fontRenderData->ItalicTextFormat().Get() : _fontRenderData->DefaultTextFormat().Get();
-    _fontInUse = drawingContext->useItalicFont ? _fontRenderData->ItalicFontFace().Get() : _fontRenderData->DefaultFontFace().Get();
+
+    DWRITE_FONT_WEIGHT weight = _fontRenderData->DefaultFontWeight();
+    DWRITE_FONT_STYLE style = _fontRenderData->DefaultFontStyle();
+    DWRITE_FONT_STRETCH stretch = _fontRenderData->DefaultFontStretch();
+
+    if (drawingContext->useItalicFont)
+    {
+        style = DWRITE_FONT_STYLE_ITALIC;
+    }
+
+    _formatInUse = _fontRenderData->TextFormatWithAttribute(weight, style, stretch).Get();

```

Comment:
After some consideration I decided to refactor this using `std::unordered_map`, which makes it super easy to add new attributed font. This is how **bold** is done:

```cpp
if (drawingContext->useBoldFont)
{
    weight = DWRITE_FONT_WEIGHT_BOLD;
}

if (drawingContext->useItalicFont)
{
    style = DWRITE_FONT_STYLE_ITALIC;
}

_formatInUse = _fontRenderData->TextFormatWithAttribute(weight, style, stretch).Get();
_fontInUse = _fontRenderData->FontFaceWithAttribute(weight, style, stretch).Get();

```

Reply:
I think this is really close to @DHowett 's vision in https://github.com/microsoft/terminal/pull/8580#issuecomment-744101996, if I understand it correctly.